### DISCUSSION TITLE
Linking files into folders

### DISCUSSION THREAD

**Discussion by JosephGallagher2001** [community/community#23768] [Other Feature Feedback, Questions, & Ideas]
<p>Hello, I am fairly new to GitHub, and I am trying to link a file into my index.html folder, such that there will be a way to access my README.md file from a link on my index page. Can someone please walk me through the steps of how to accomplish this? If someone could, that would be great.<br>
Below is my README.md file link: <a href="https://github.com/JosephGallagher2001/JosephGallagher2001.github.io/blob/master/README.md" rel="nofollow noopener">https://github.com/JosephGallagher2001/JosephGallagher2001.github.io/blob/master/README.md</a><br>
Best, JosephGallagher2001</p>

**Comment by darrowby385** [comment, marked as answer]
<p>Hi, <a class="mention" href="/josephgallagher2001">@JosephGallagher2001</a>!</p>
<p>Before I can answer your question about creating links, it’s important to clarify some details about how GitHub Pages publishes files, particularly <code>README.md</code> and <code>index.html</code>.</p>
<h3>An important detail about <code>README.md</code> and <code>index.html</code>
</h3>
<p>When you publish a repository as a GitHub Pages site, the GitHub Pages build process converts Markdown files to HTML, for viewing on the web. These converted files aren’t stored in your repository, as they only exist as part of your published website. Any HTML files in your repository are not converted. Instead, they are published exactly as-is.</p>
<p>For example, if you had a Markdown file in your repository named <code>about.md</code>, GitHub Pages would convert it to HTML and publish it at <code>JosephGallagher2001.github.io/about.html</code>.</p>
<p>If you had an HTML file in your repository named <code>blog.html</code>, GitHub Pages would publish it as-is, at <code>JosephGallagher2001.github.io/blog.html</code>.</p>
<p>As you can see, the published URL for a page generally matches its filename, with <code>.md</code> becoming <code>.html</code>. There is, however, an exception to this rule: GitHub Pages will not publish <code>README.md</code> as <code>README.html</code>, but rather as <code>index.html</code>. It does this because <code>index.html</code> is an accepted standard for a site’s default landing page, and <code>README.md</code> commonly serves a similar purpose on GitHub.</p>
<p>Your repository contains both a <code>README.md</code> file and an <code>index.html</code> file. This presents a conflict, as GitHub Pages is designed to publish both files as <code>index.html</code>. Under these circumstances, the <code>index.html</code> file in your repository takes precedence, and GitHub Pages skips the process of converting your <code>README.md</code> file to HTML.</p>
<p>All of this is to say that you can’t create a link between an <code>index.html</code> file and <code>README.md</code>, due to the unique publishing conflict relevant to these files. You can, however, create links between any other files in your repository.</p>
<h3>Okay, now we can talk about links!</h3>
<p>As an example, let’s say you modified your repository to contain two Markdown files: <code>index.md</code> and <code>about.md</code>. To link from the first to the second, you can use this syntax:</p>
<pre><code class="lang-auto">[Click here to visit my about page!](about.md)
</code></pre>
<p>GitHub Pages will then convert these Markdown files to HTML, publishing them as <code>index.html</code> and <code>about.html</code>. In this process, it will convert the above Markdown syntax into valid HTML. The HTML syntax for this link would look like this:</p>
<pre><code class="lang-auto">&lt;a href="about.html"&gt;Click here to visit my about page!&lt;/a&gt;
</code></pre>
<p>If you want to keep a mix of HTML and Markdown files in your repository, you’ll need to use the HTML syntax when creating links in your HTML documents, and the Markdown syntax when creating links in your Markdown documents. Be careful to note the different file extensions.</p>
<p>I hope that helps! If you have any follow-up questions, please don’t hesitate to ask.</p>
<p>Friday</p>

**Comment by Samuel-Tham65** [comment]
i have a certain issue that i cant seem to fix right now i have link on my index page that i puloaded but when i try to open the  link not only does it not start with https but start with file/// i cant even open the link
